tests/cupy_tests/manipulation_tests/test_kind.py

```diff
@@ -10,6 +10,25 @@
 @testing.gpu
 class TestKind(unittest.TestCase):
 
+    @testing.for_orders('CFAK')
+    @testing.for_all_dtypes()
+    @testing.numpy_cupy_array_equal()
+    def test_asarray_chkfinite(self, xp, dtype, order):
+        a = [0, 4, 0, 5]
+        return xp.asarray_chkfinite(a, dtype=dtype, order=order)
+
+    @testing.for_orders('CFAK')
+    @testing.for_all_dtypes()
```

Comment:
It does not make sense to test bool with nan/inf, so we simply disable it. (Alternatively, we can use `contextlib.nullcontext` for bool, which is available since Python 3.7, and a `pytest.raise` context manager for other dtypes. I don't have strong opinions here.) 
```suggestion
    @testing.for_all_dtypes(no_bool=True)
```